WireAPI/Sources/WireAPI/Authentication/PersistentAuthenticationStorage.swift

```diff
@@ -0,0 +1,192 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireFoundation
+
+public actor PersistentAuthenticationStorage: AuthenticationStorage {
+
+    private typealias Error = PersistentAuthenticationStorageError
+
+    private let userID: UUID
+    private var accessToken: AccessToken?
+    private let sharedUserDefaults: UserDefaults
+    private let keychain: any KeychainProtocol
+    private static let cookieEncryptionKeyKey = "ZMCookieKey"
+
+    public init(
+        userID: UUID,
+        sharedUserDefaults: UserDefaults,
+        keychain: any KeychainProtocol
+    ) {
+        self.userID = userID
+        self.sharedUserDefaults = sharedUserDefaults
+        self.keychain = keychain
+    }
+
+    // MARK: - Access token
+
+    public func storeAccessToken(_ accessToken: AccessToken) async {
+        self.accessToken = accessToken
+    }
+
+    public func fetchAccessToken() async -> AccessToken? {
```

Comment:
Like @jullianm implied please check that these `async` is needed here an in other places. I was under the impression that an actor can conform to the async protocol definitions without it.

Reply:
If I remove `async` it won't compile because of actor isolation. And if the protocol methods are not async but he actor methods are, then the actor won't conform to the protocol.

Reply:
@johnxnguyen Have you actually tried this. For me the following compiles which makes me think the actor is conforming to the protocol despite missing the `async` call:

```swift
protocol Animal {
    func makeNoise() async -> String
}

actor Dog: Animal {
    func makeNoise() -> String {
        "Woof!"
    }
}
```